java/src/main/native/OrtJniUtil.c

```diff
@@ -278,8 +330,72 @@ jobject createEmptyMapInfo(JNIEnv *jniEnv) {
+    ONNXType type;
+    checkOrtStatus(jniEnv,api,api->GetOnnxTypeFromTypeInfo(elementTypeInfo,&type));
+
+    jobject sequenceInfo;
+
+    switch (type) {
+        case ONNX_TYPE_TENSOR: {
+            // Figure out element type
+            const OrtTensorTypeAndShapeInfo* elementTensorInfo;
+            checkOrtStatus(jniEnv,api,api->CastTypeInfoToTensorInfo(elementTypeInfo,&elementTensorInfo));
+            ONNXTensorElementDataType element;
+            checkOrtStatus(jniEnv,api,api->GetTensorElementType(elementTensorInfo,&element));
+
+            // Convert element type into ONNXTensorType
+            jint onnxTypeInt = convertFromONNXDataFormat(element);
+            // Get the ONNXTensorType enum static method
+            char *onnxTensorTypeClassName = "ai/onnxruntime/TensorInfo$OnnxTensorType";
+            jclass onnxTensorTypeClazz = (*jniEnv)->FindClass(jniEnv, onnxTensorTypeClassName);
+            jmethodID onnxTensorTypeMapFromInt = (*jniEnv)->GetStaticMethodID(jniEnv,onnxTensorTypeClazz, "mapFromInt", "(I)Lai/onnxruntime/TensorInfo$OnnxTensorType;");
+            jobject onnxTensorTypeJava = (*jniEnv)->CallStaticObjectMethod(jniEnv,onnxTensorTypeClazz,onnxTensorTypeMapFromInt,onnxTypeInt);
+
+            // Get the ONNXJavaType enum static method
+            char *javaDataTypeClassName = "ai/onnxruntime/OnnxJavaType";
+            jclass onnxJavaTypeClazz = (*jniEnv)->FindClass(jniEnv, javaDataTypeClassName);
+            jmethodID onnxJavaTypeMapFromONNXTensorType = (*jniEnv)->GetStaticMethodID(jniEnv,onnxJavaTypeClazz, "mapFromOnnxTensorType", "(Lai/onnxruntime/TensorInfo$OnnxTensorType;)Lai/onnxruntime/OnnxJavaType;");
+            jobject onnxJavaType = (*jniEnv)->CallStaticObjectMethod(jniEnv,onnxJavaTypeClazz,onnxJavaTypeMapFromONNXTensorType,onnxTensorTypeJava);
+
+            // Construct sequence info
+            jmethodID sequenceInfoConstructor = (*jniEnv)->GetMethodID(jniEnv,sequenceInfoClazz,"<init>","(ILai/onnxruntime/OnnxJavaType;)V");
+            sequenceInfo = (*jniEnv)->NewObject(jniEnv,sequenceInfoClazz,sequenceInfoConstructor,(jint)-1,onnxJavaType);
+            break;
+        }
+        case ONNX_TYPE_MAP: {
+            // Extract the map info
+            const OrtMapTypeInfo* mapInfo;
+            checkOrtStatus(jniEnv,api,api->CastTypeInfoToMapTypeInfo(elementTypeInfo,&mapInfo));
+
+            // Convert it using the existing convert function
+            jobject javaMapInfo = convertToMapInfo(jniEnv,api,mapInfo);
+
+            // Construct sequence info
+            jmethodID sequenceInfoConstructor = (*jniEnv)->GetMethodID(jniEnv,sequenceInfoClazz,"<init>","(ILai/onnxruntime/MapInfo;)V");
+            sequenceInfo = (*jniEnv)->NewObject(jniEnv,sequenceInfoClazz,sequenceInfoConstructor,(jint)-1,javaMapInfo);
+            break;
+        }
+        default: {
+            sequenceInfo = createEmptySequenceInfo(jniEnv);
```

Comment:
Is this line needed if you're throwing an exception any way?

Reply:
I don't think so. It might be there to prevent an unused variable warning. I'll check

Reply:
It doesn't seem to prevent any warnings on my Mac, but I've had different compiler warnings than the builds before. I can set it to NULL if you want?